woocommerce/woocommerce
# Use WP reset_password() function to reset the customer password

### All Submissions:






### Changes proposed in this Pull Request:

Currently, there is a custom implementation of `WC_Shortcode_My_Account::reset_password()` function which does not fire `after_password_reset` hook after the password has been reset so there is no way for external plugins to listen for that change. The native implementation is better to use because it fires the hook and also sets the `default_password_nag` option to `false`.

Generally, I think it's better to use native functions because it ensures that we are following the latest changes in the WP Core.

Particularly, this PR adds the `after_password_reset` hook support so external plugins could listen for it.

Note: I couldn't find test files that would be responsible for this part so I haven't created any tests. If there is any place where I should add a test, please point me there.

### How to test the changes in this Pull Request:

1. Add custom action hook listening for a password reset
```php
add_action( 'after_password_reset', 'my_customer_password_reset', 10, 2 );
function my_customer_password_reset( $user, $new_pass) {
    die('The hook has been fired');
}
```
2. Open the 'lost-password' page and request the password reset link
3. Change your password to the new one
4. The `my_customer_password_reset()` function should be fired after the password change

### Other information:




### Changelog entry

> Use WP reset_password() function to reset the customer password so the `after_password_reset` hook will be fired
